multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java

```diff
@@ -209,21 +203,15 @@ private CloudServiceInstanceExtended setServiceParameters(ProcessContext context
         String spaceGuid = context.getVariable(Variables.SPACE_GUID);
         return fileService.processFileContent(spaceGuid, appArchiveId, appArchiveStream -> {
             try (InputStream is = ArchiveHandler.getInputStream(appArchiveStream, fileName, configuration.getMaxManifestSize())) {
-                return mergeCredentials(service, is);
+                Map<String, Object> credentials = JsonUtil.convertJsonToMap(is);
+                return ImmutableCloudServiceInstanceExtended.copyOf(service)
+                                                            .withCredentials(credentials);
```

Comment:
we still have to merge the service parameters from the mtad.yaml and extension descriptors as before, not replacing them entirely with the json file.
https://help.sap.com/docs/btp/sap-business-technology-platform/service-creation-parameters
under
parameters:
  config:
     ...